cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -83,7 +83,7 @@ pub struct PathEnvironment {
 /// A profile script or list of packages to install when initializing an environment
 #[derive(Debug, Default, Eq, Ord, PartialEq, PartialOrd)]
 pub struct InitCustomization {
-    pub profile: Option<String>,
+    pub script: Option<String>,
```

Comment:
As you highlight any setting of aliases must be repeated with each new subshell, so with that it follows that aliases can only be set in the appropriate `[profile.<shell>]` section, because the `[hook.on-activate]` script is only invoked once.

That leaves us with only two options:
1. if the script that emits alias configuration statements is idempotent then the user can simply `eval $(script ...)` from within `[profile.<shell>]` (the easy, preferred way, but again that depends on the script)
2. otherwise the user will need to save the output from the script as run in `[hook.on-activate]`, and then eval the results of that invocation from within `[profile.<shell>]`, and our ability to set and pass environment variables down to the userShell will be crucial for making that happen

But the key consideration to me is that flox environments are designed for use with multiple shells, and users therefore need for the `[hook.on-activate]` to work in a way that it can be invoked once for the benefit of all other shell types to follow, and that simply cannot happen if the userShell is the thing running the hook. In that regard `on-activate` hooks are as early in the activation process as you can get, performing actions and preparing an environment to be inherited by the user shells that follow.

Reply:
Yeah we are hitting the same points in multiple threads of conversation. My latest response is [here](https://github.com/flox/flox/pull/1299/#discussion_r1571934240) - closing out this conversation in favour of that one.